server/src/controller/todoItemController.ts

```diff
@@ -0,0 +1,64 @@
+import { todoItemRepo } from '../repositories/itemRepository';
+import { TodoItem } from '../models/todoItem';
+import { InvalidParameterError, NotFoundError } from '../common/errors';
+
+interface TodoItemDTO {
+  title: string;
+  description: string;
+  datecrated: Date;
+  dateDue?: Date;
+  isResolved: boolean;
+  priority: number;
+}
+
+export class TodoItemController {
+  static getItems = async (): Promise<TodoItem[]> => {
+    return await todoItemRepo.find();
+  };
+
+  static getItemById = async (id: number): Promise<TodoItem> => {
+    const item = await todoItemRepo.findOneBy({ id });
```

Comment:
Usually the controller validates the request and parses the response. Id is no being validated. The controller layer call the services layer. The service layer calls the repository layer.

Reply:
Added ID validation